mandible/metadata_mapper/storage.py

```diff
@@ -15,12 +16,41 @@ class StorageError(Exception):
 STORAGE_REGISTRY: Dict[str, Type["Storage"]] = {}
 
 
-@dataclass
 class Storage(ABC):
     # Registry boilerplate
-    def __init_subclass__(cls):
-        STORAGE_REGISTRY[cls.__name__] = cls
+    def __init_subclass__(cls, register: bool = True):
+        if register:
+            STORAGE_REGISTRY[cls.__name__] = cls
+
+    @abstractmethod
+    def open_file(self, context: Context) -> IO[bytes]:
+        """Get a filelike object to access the data."""
+        pass
+
+
+@dataclass
+class Dummy(Storage):
+    """A dummy storage that returns a hardcoded byte stream.
 
+    Used for testing.
+    """
+
+    data: Union[str, bytes]
+
+    def open_file(self, context: Context) -> IO[bytes]:
+        if isinstance(self.data, str):
+            data = self.data.encode()
+        else:
+            data = self.data
+
+        return io.BytesIO(data)
+
+
+@dataclass
+class FilteredStorage(Storage, register=False):
```

Comment:
Should inherit from `ABC`. Maybe name all unregistered classes starting with `_` as well?

Reply:
Since `Storage` inherits from `ABC`, `FilteredStorage` doesn't need to. Going to leave it as is for now.